### PR TITLE
DOC: AWS secret manager

### PR DESCRIPTION
**Pull request by Shubham9t9**
# Description

Documentation for adding secrets using AWS secret manager.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# Checklist:

* [X] The title of the PR states what changed and the related issues number (used for the release note).
* [X] Does this PR require documentation updates?
* [X] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas
* [ ] I have tested it for all user roles

### CODE REVIEW

File: docs/user-guide/creating-application/secrets.md [hunk 1]
```diff
@@ -72,3 +72,56 @@ You can delete your secret. Click on your secret and click on the `delete sign`
 
 ![](../../.gitbook/assets/delete_secret%20%282%29.png)
 
+## Add External secrets
```
Comment (prakarsh-dt) [inline docs/user-guide/creating-application/secrets.md:75]: Change to:
External Secrets

Comment (Shubham9t9) [reply docs/user-guide/creating-application/secrets.md]: changed

File: docs/user-guide/creating-application/secrets.md [hunk 2]
```diff
@@ -50,25 +39,102 @@ File permission will be provide at the configmap level not on the each key of th
...
+The secret that is already created and stored in the environment and being used by devtron externally is referred here as `Kubernetes External Secret`. For this option, devtron will not create any secret by itself but they can be used within the pods. Before adding secret from kubernetes external secret, please make sure that secret with the same name is present in the environment. To add secret from kubernetes external secret, follow the steps mentioned below:
+
+1. Navigate to `Secrets` of the application.
+2. Click on `Add Secret` to add a new secret.
+3. Select `Kubernetes External Secret` from dropdown of `Data type`.
+4. Provide a name to your secret. Devtron will search secret in the environment with the same name that you mention here. 
+
+### AWS Secret Manager
+
+Before adding any external secrets on devtron, `kubernetes-external-secrets` must be installed on devtron. Kubernetes External Secrets allows you to use external secret management systems (e.g., AWS Secrets Manager) to securely add secrets in Kubernetes.
```
Comment (prakarsh-dt) [inline docs/user-guide/creating-application/secrets.md:91]: ...must be installed on the target cluster.

(e.g., AWS Secrets Manager, Hashicorp Vault etc)

Comment (Shubham9t9) [reply docs/user-guide/creating-application/secrets.md]: done

**Review by prakarsh-dt — APPROVED** [review]

**Pull request merged**